bazel/repository.bzl

```diff
@@ -28,6 +28,17 @@ def opentelemetry_cpp_deps():
         ],
     )
 
+    # Load abseil dependency(optional)
+    maybe(
+        http_archive,
+        name = "com_google_absl",
+        sha256 = "59b862f50e710277f8ede96f083a5bb8d7c9595376146838b9580be90374ee1f",
+        strip_prefix = "abseil-cpp-20210324.2",
```

Comment:
abseil version doesn't match with what we use in CI: [20210324.0](https://github.com/owent-contrib/opentelemetry-cpp/blob/3a8157649c8ab0fb147252ce65ecce8ae4046b9c/ci/install_abseil.sh#L11)

Reply:
Fixed. The version of internal abseil in `api/include/opentelemetry/nostd/absl` is `v2020-03-03#8` ,should it be upgraded?

Reply:
> The version of internal abseil in api/include/opentelemetry/nostd/absl is v2020-03-03#8, should it be upgraded?

One of the concerns I see with the periodic upgrade of the internal `nostd::variant` snapshot would be the ABI incompatibility issue if the instrumented library and SDK are built using different otel-cpp releases (with different `nostd::variant` snapshots).